lld/MachO/SymbolTable.cpp

```diff
@@ -91,6 +97,11 @@ static void transplantSymbolsAtOffset(InputSection *fromIsec,
     }
     return true;
   });
+  assert(llvm::is_sorted(toIsec->symbols,
+                         [](const Defined *s, const Defined *t) {
+                           return s->value < t->value;
+                         }) &&
+         "Symbols should still be sorted at exit.");
```

Comment:
I'll probably get rid of this one, since I can't get it to fire. My theory is that the failure of this assertion can only be a consequence of UB, so the compiler just optimises this away. (No real evidence to support this though, so could be wrong!)

Reply:
Does `is_partitioned` trigger without expensive checks ? On which tests does it trigger ?

Reply:
I've verified that `is_partitioned` triggers _with_ expensive checks with Clang, and _without_ expensive checks with GCC.

The failing tests (for both) are
```
Failed Tests (3):
  lld :: MachO/local-alias-to-weak.s
  lld :: MachO/symtab.s
  lld :: MachO/weak-definition-gc.s
```

I haven't yet verified the assertion failure for Clang without expensive checks. The build is running now; I'll report back soon.

Reply:
Verified with Clang _without_ expensive checks and the comparison in `symSucceedsOff` reversed:
```
Failed Tests (3):
  lld :: MachO/local-alias-to-weak.s
  lld :: MachO/symtab.s
  lld :: MachO/weak-definition-gc.s
```
To be sure, all errors are of the form
```
Assertion failed: (std::is_partitioned(toIsec->symbols.begin(), toIsec->symbols.end(), [toOff, symSucceedsOff](const Symbol *s) { return !symSucceedsOff(toOff, s); }) && "Symbols in toIsec must be partitioned by toOff."), function transplantSymbolsAtOffset, file SymbolTable.cpp, line 78.
```

Reply:
Great, thanks for checking this.